web/src/components/workflow-editor/edit-panel/action-edit-panel.tsx

```diff
@@ -134,20 +160,62 @@ export default function ActionEditPanel() {
 									<Flex>
 										<Code>{secretPlaceholder}</Code>
 									</Flex>
-									<TextField.Root
-										variant="surface"
+									<Select.Root
 										value={
 											actionData.secretsMapping[
 												secretPlaceholder
 											]
 										}
-										onChange={(event) =>
+										onValueChange={(value) =>
 											onChangeSecretsMapping(
 												secretPlaceholder,
-												event,
+												value,
 											)
 										}
-									/>
+									>
+										<Select.Trigger />
+										<Select.Content>
+											<Select.Group>
+												<Select.Label>
+													Select a secret
+												</Select.Label>
+												{secrets.map((_, idx) => (
+													<Select.Item
+														key={
+															secrets[idx]
+																.secretId
+														}
+														value={
+															secrets[idx]
+																.secretId
+														}
+													>
+														{secrets[idx].secretId}
+													</Select.Item>
+												))}
+												<Select.Separator />
+												<Select.Item value="$$$save_and_redirect$$$">
+													Save Workflow and Redirect
+													to Secrets
+												</Select.Item>
+											</Select.Group>
+										</Select.Content>
+									</Select.Root>
+									{hasDeletedSecret(
+										action.id,
+										secretPlaceholder,
+									) && (
+										<Flex
+											direction="row"
+											align="center"
+											className="text-red-500 font-medium mt-2"
```

Comment:
If the component has props for doing what you want (text color, font weight), then please use this instead of using tailwindcss. You can just do `<Text color="red" weight="medium">` for example.

Reply:
Also, `<Flex>` has props for margin top: `<Flex mt="2">`.